lib/utils/platform_database_factory.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:path/path.dart' as path;
+import 'package:path_provider/path_provider.dart';
+import 'package:sembast/sembast.dart';
+import 'package:sembast/sembast_io.dart';
+import 'package:sembast/sembast_memory.dart';
+import 'package:sembast_web/sembast_web.dart';
+
+class PlatformDatabaseFactory extends DatabaseFactory {
+  String _databaseName = '';
```

Comment:
A bit dangerous that you need to set the database name, otherwise it will just use `.json`. Maybe use a default or init with null and check if it was set in `_finalPath()`?